crates/core/main.rs

```diff
@@ -79,17 +79,12 @@ fn try_main(args: Args) -> Result<()> {
 fn search(args: &Args) -> Result<bool> {
     let started_at = Instant::now();
     let quit_after_match = args.quit_after_match()?;
-    let subject_builder = args.subject_builder();
     let mut stats = args.stats()?;
     let mut searcher = args.search_worker(args.stdout())?;
     let mut matched = false;
     let mut searched = false;
-
-    for result in args.walker()? {
-        let subject = match subject_builder.build_from_result(result) {
-            Some(subject) => subject,
-            None => continue,
-        };
+    let subjects = sorted_results(args);
+    for subject in subjects {
```

Comment:
This is a big no-no from what I can tell. It looks like you're _unconditionally_ collecting all paths before iterating over them even if no `--sort` flag is passed. That means you're incurring the perf/memory/latency---at least in part---of sorting even when you don't need to.

Reply:
resolved by https://github.com/BurntSushi/ripgrep/pull/2361/commits/d4706913adf88cb3c6b395d0afcf493f39208169